org.eclipse.dawnsci.remotedataset.client/src/org/eclipse/dawnsci/remotedataset/client/RemoteDataHolder.java

```diff
@@ -0,0 +1,193 @@
+package org.eclipse.dawnsci.remotedataset.client;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.dawnsci.analysis.api.io.IDataHolder;
+import org.eclipse.dawnsci.analysis.api.io.IFileLoader;
+import org.eclipse.dawnsci.analysis.api.persistence.IMarshallerService;
+import org.eclipse.dawnsci.analysis.api.tree.Tree;
+import org.eclipse.dawnsci.analysis.tree.TreeToMapUtils;
+import org.eclipse.dawnsci.remotedataset.XMLMarshallerService;
+import org.eclipse.january.dataset.IDataset;
+import org.eclipse.january.dataset.ILazyDataset;
+import org.eclipse.january.dataset.LazyDynamicDataset;
+import org.eclipse.january.io.ILazyLoader;
+import org.eclipse.january.metadata.IMetadata;
+
+public class RemoteDataHolder implements IDataHolder {
+	
+	private String path;
+	private String host;
+	private int port;
+	private URLBuilder urlBuilder;
+	private Tree tree;
+	
+	
+	
+	public RemoteDataHolder(String path, String host, int port) {
+		this.urlBuilder = new URLBuilder(host, port);
+		urlBuilder.setPath(path);
+		try {
+			this.tree = TreeToMapUtils.mapToTree(getTreeMap(),path);
+		} catch (Exception e) {
+			// TODO Auto-generated catch block
+		}
+	}
+
+	private Map<String, Object> getTreeMap() throws Exception {
+		
+		urlBuilder.setGet(true);
+		final URL url = new URL(urlBuilder.getTreeURL());
+		URLConnection  conn = url.openConnection();
+
+		final BufferedReader reader = new BufferedReader(new InputStreamReader(conn.getInputStream(), "UTF-8"));
+		try {
+			String xml = reader.readLine();
+			IMarshallerService marshaller = new XMLMarshallerService();
+			return (Map<String, Object>)marshaller.unmarshal(xml, Map.class);
+
+		} finally {
+			reader.close();
+		}
+	}
+	
+	@Override
+	public IDataHolder clone() {
+		return null;
+	}
+
+	@Override
+	public IMetadata getMetadata() {
+		return null;
+	}
+
+	@Override
+	public void setMetadata(IMetadata md) {
+
+	}
+
+	@Override
+	public IDataset getDataset(int index) {
+		return null;
+	}
+
+	@Override
+	public IDataset getDataset(String name) {
+		return null;
+	}
+
+	@Override
+	public ILazyDataset getLazyDataset(int index) {
+		return null;
+	}
+
+	@Override
+	public ILazyDataset getLazyDataset(String name) {
+		URLBuilder b = new URLBuilder(urlBuilder);
+		b.setDataset(name);
+		RemoteLoader loader = new RemoteLoader(b);
+		try {
+			RemoteDatasetInfo datasetInfo = loader.getDatasetInfo();
+			LazyDynamicDataset lazy = new LazyDynamicDataset(name, datasetInfo.getDtype(),
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "lazy". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)